Adding dave2d driver support on lvgl for Renesas RA devices

duynguyenxa opened the pull request:
This PR introduces support for D/AVE 2D (Dave2D) hardware acceleration in LVGL on Zephyr OS.
The integration allows LVGL’s rendering engine to utilize the Renesas D/AVE 2D GPU for improved graphics performance, enabling faster UI updates and smoother animations on supported hardware platforms (e.g., RA8D1, RA8P1).

duynguyenxa committed:
manifest: Update revision for renesas hal and lvgl

This commit update commit ID for Renesas hal module
to support dave2d library

Signed-off-by: Duy Nguyen <duy.nguyen.xa@renesas.com>

duynguyenxa committed:
drivers: misc: Initial support for drw driver on Renesas RA

First commit to add support for Renesas RA drw driver

Signed-off-by: Duy Nguyen <duy.nguyen.xa@renesas.com>

duynguyenxa committed:
dts: arm: renesas: Add DRW nodes for RA8x1 and RA8x2 MCU

Add DRW nodes to ra8x1.dtsi and ra8x2.dtsi, add delete-node for
unsupported RA8 devices like RA8T1, RA8M1, RA8T2, RA8M2

Signed-off-by: Duy Nguyen <duy.nguyen.xa@renesas.com>

uLipe commented:
Thank you for bringing this one @duynguyenxa, I was about to ask you about the progress on this.

JarmouniA reviewed (commented):
The only Dave2D I know is https://m.youtube.com/channel/UCVYamHliCI9rw1tHR1xbkfw :)

uLipe reviewed (dismissed):
Great work! It will simplify a lot the Weather Demo from Renesas under Zephyr.

faxe1008 commented on west.yml:
```diff
@@ -313,7 +313,7 @@ manifest:
       path: modules/lib/loramac-node
     - name: lvgl
-      revision: b03edc8e6282a963cd312cd0b409eb5ce263ea75
+      revision: pull/67/head
```
I intended to open a PR upstream to incorporate this, but from what I can see this should be resolved now:
https://github.com/zephyrproject-rtos/lvgl/pull/67#issuecomment-3595068190

So as soon as the module is updated this might be dropped as well.

faxe1008 commented on samples/modules/lvgl/demos/sample.yaml:
```diff
@@ -89,6 +93,8 @@ tests:
       - SHIELD=rtklcdpar1s00001be
     extra_configs:
-      - CONFIG_LV_DEMO_MUSIC_LANDSCAPE=y
+      - CONFIG_LV_Z_DEMO_WIDGETS=y
```
Not bent on any particular demo, but any reason to use the widgets one specifically?

faxe1008 reviewed (commented):
LVGL part looks fine, will +1 if the `hal` PR is done

duynguyenxa committed:
modules: lvgl: Add glue config for LV_ATTRIBUTE_MEM_ALIGN_SIZE

The DAVE2D hardware require data buffer alignment setting so that
hardware can work properly, so this add LV_ATTRIBUTE_MEM_ALIGN_SIZE glue
config to lvgl config

Signed-off-by: Duy Nguyen <duy.nguyen.xa@renesas.com>

duynguyenxa commented:
Hi @faxe1008 , could you take a look at this, hal branch is already merged

duynguyenxa committed:
boards: renesas: Add support for drw driver on RA8

Add support for drw driver on ek_ra8p1 and ek_ra8d1

Signed-off-by: Duy Nguyen <duy.nguyen.xa@renesas.com>

duynguyenxa committed:
boards: shields: adjust GLCDC panel timing parameters

Adjust GLCDC panel timing parameters for
rtklcdpar1s00001be display shield and add missing config for lvgl when
used with DAVE2D

Signed-off-by: Duy Nguyen <duy.nguyen.xa@renesas.com>

duynguyenxa commented:
LVGL and hal_rensas maintainer approve for the use of DAVE2D binary blob library, so I  removed the tag
